### PR TITLE
CDPS-147: Case Notes Mop Up

### PR DESCRIPTION
**Pull request by davidmiddletonmoj**
CDPS-150, CDPS-152, CDPS-153, CDPS-163

### CODE REVIEW

File: architecture-decision-record/0004-use-user-token-for-case-notes-api.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# 3. Use `async`/`await` pattern over `Promise.then`
```
Comment (danielburnley) [inline architecture-decision-record/0004-use-user-token-for-case-notes-api.md:1]: 👀 need to update the title

File: server/utils/roleHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Role } from '../data/enums/role'
+
+// eslint-disable-next-line import/prefer-default-export
+export const canViewOrAddCaseNotes = (roles: string[], activeCaseLoadId: string, prisonId: string) => {
+  return (
+    (roles?.some(role => role === Role.InactiveBookings) && ['OUT', 'TRN'].includes(prisonId)) ||
+    activeCaseLoadId === prisonId
+  )
+}
```
Comment (danielburnley) [inline server/utils/roleHelpers.ts:9]: Just to confirm that the intended behaviour is as long as it's your active case load and not "a case load that you have"? I'm assuming it's right but just double checking :D

**Review by danielburnley — APPROVED** [review]
Just a couple questions but once you've updated that title in the ADR feel free to merge 🎉

File: server/controllers/overviewController.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { Request, Response } from 'express'
```
[the remark this answers is not in the document]
Comment (davidmiddletonmoj) [reply server/controllers/overviewController.ts:1]: Yeah... routes had no tests - just transposed the logic to the controller to make things easier to read with the added role stuff.
Will create a task to do testing later on.

**Pull request merged**